db/migrate/20181206123910_create_foreman_tasks_links.foreman_tasks.rb

```diff
@@ -0,0 +1,28 @@
+class CreateForemanTasksLinks < ActiveRecord::Migration[4.2]
+  def change
+    # rubocop:disable Rails/CreateTableWithTimestamps
+    create_table :foreman_tasks_links do |t|
+      task_id_options = { :index => true, :null => true }
+      if on_postgresql?
+        t.uuid :task_id, task_id_options
+      else
+        t.string :task_id, task_id_options
+      end
+      # TODO: Figure out if we need this
+      # t.string :name, index: true, null: false
+      t.string :resource_type
+      t.integer :resource_id
+    end
+    # rubocop:enable Rails/CreateTableWithTimestamps
+
+    add_index :foreman_tasks_links, [:resource_type, :resource_id]
+    add_index :foreman_tasks_links, [:task_id, :resource_type, :resource_id],
```

Comment:
I wonder if we need the unique constraint here - just thinking if it doesn't influence negatively the link creation process once there are many of those?

Reply:
My expectation was a task can be linked to a resource only once and having it enforced by the database seemed like the right thing to do. If we decided to go the other way and check if a link exists in ruby, then we'd still need this index to be able to find the record quickly. At that point, there is probably no difference between an index and an index with unique constraint. Or we could just not check it at all and live with occasional duplicates

Reply:
That's true. One options might be to cache the task object on `Thread.current` in https://github.com/theforeman/foreman-tasks/blob/master/app/lib/actions/base.rb#L9, in which case all the plannings would know which resources are already linked, and the deduplication could be achieved in memory. I'm ok though to have that handled as separate issue, while thinking more about it first.

Reply:
Ok, I'll leave it as it is for now.